src/cpr_data_access/utils.py

```diff
@@ -0,0 +1,67 @@
+import csv
+from pathlib import Path
+
+
+def is_sensitive_query(text: str, sensitive_terms: set) -> bool:
+    """
+    Scans text to determine if the query should be considered sensitive
+
+    It does this by evaluating it against a set of predefined sensitive terms.
+    These, as well as the specific logic are reproduced from previous work, which
+    stated that "If the query contains any sensitive terms, and the length of the
+    shortest sensitive term is >=50% of the length of the query by number of words..."
+    then it is considered sensitive. Further details on the original can be found here:
+    https://github.com/climatepolicyradar/navigator/pull/815
+
+    This updated version builds on the above to avoid a loophole where a string of
+    sensitive terms can be incorrectly flagged as not being sensitive. This happens
+    because it is the shortest sensitive term that is compared to the rest of the
+    query, and the rest of the query at that point can contain other sensitive terms.
+
+    """
+    sensitive_terms_in_query = [
+        term for term in sensitive_terms if term in text.lower()
+    ]
+
+    if sensitive_terms_in_query:
+        shortest_sensitive_term = min(sensitive_terms_in_query, key=len)
+        shortest_sensitive_word_count = len(shortest_sensitive_term.split(" "))
```

Comment:
![image](https://github.com/climatepolicyradar/data-access/assets/58440325/b78dda90-6a63-4f4d-9e98-9c1fcf65287c)

Just to be clear!

Reply:
Good spot, although I think this should be okay as these words are coming out of the tsv file. I'll double check

Reply:
Interestingly it turns out some of those words have trailing spaces, although the load function cleans them up, so seems we only end up with phrases that don't have extra spaces